pkg/cmd/pulumi/stack_init.go

```diff
@@ -161,5 +170,56 @@ func newStackInitCmd() *cobra.Command {
 		&stackToCopy, "copy-config-from", "", "The name of the stack to copy existing config from")
 	cmd.PersistentFlags().BoolVar(
 		&noSelect, "no-select", false, "Do not select the stack")
+	cmd.PersistentFlags().StringArrayVar(&teams, "teams", nil, "A list of team "+
+		"names that should have permission to read and update this stack,"+
+		" once created")
 	return cmd
 }
+
+// This function constructs a createStackOptions object if
+// valid, otherwise returning nil. Most backends expect nil
+// options, and error if options are non-nil.
+func validateCreateStackOpts(stackName string, b backend.Backend, teams []string) (backend.CreateStackOptions, error) {
```

Comment:
I do wonder if this should all just be in the backend? Just kill "SupportsTeams", and return the teamsUnsupportedError from filestate rather than internalIllegalTeamsError.

Reply:
I decided to perform validation in `cli` instead of in the `backend` to adhere to the "fail fast" methodology. Ideally, we should error as soon as we can determine user input is invalid. This has a few benefits (especially in this case):

It removes the possibility of "poisoned" values from floating around the engine. In this case, having a `LocalBackend` is mutually exclusive with having Teams in `StackCreateOptions`. Instead of creating an illegal state (i.e. those two structs coexisting), we can potentially prevent errors by eagerly pruning illegal values. Further, the backend can more boldly assert the correctness of its invariants; it can expect arguments to be well-formed. It doesn't need to re-validate data. Practically, this means upgrading the fatality of illegal state.

You're right that it adds a little complexity by requiring `SupportsTeams` and `internalIllegalTeamsError` but I'd argue that's a limitation of the expressiveness of Go. We're _like_ the type of concrete type of `CreateStackOpts` to be dependant on the concrete type of `Backend`. We could enforce that at compile-time if Go had path-dependent types, or maybe with a fancy combination of multiple generics, but that's far afield of the current API.